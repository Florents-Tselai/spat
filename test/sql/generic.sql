SET client_min_messages = ERROR;

SELECT spat_db_name(); -- spat-default
-- switch to a new db
SET spat.db = 'test-spat';
SELECT spat_db_name(); -- test-spat
SELECT spat_db_created_at() NOTNULL;

-- switch back to the default to preserve my sanity when copy-pasting

SET spat.db = 'spat-default';

/* -------------------- STRINGS / GENERIC -------------------- */

-- SET - DELETE - SIZE
SELECT spset('tkey1', 'dfgf');
SELECT spset('tkeygdfg1', 'dfgf');
SELECT spset('gdf', 'dfgf');
SELECT spset('gfd', 'dfgf');

SELECT sp_db_nitems(); --4

-- DELETE and inspect size
SELECT del('tkey1'); --existing
SELECT del('aaaa'); --not existing
SELECT sp_db_nitems(); --3
SELECT del('tkeygdfg1');
SELECT del('gdf');
SELECT del('gfd');
SELECT sp_db_nitems(); --0

-- SET / GET / DEL text
SELECT spset('key1', 'value1');
SELECT spget('key1');

SELECT del('key1');
SELECT spget('key1');

-- TTL

SELECT SPSET('expkey1', 'expvalue1', ttl=> '1 second');
SELECT ttl('expkey1') < '1 second' ;

-- DB SIZE BYTES
SELECT sp_db_size_bytes();
SELECT sp_db_size();

-- sptype
SELECT spset('key1', 'value1');
SELECT sptype('key1');

SELECT sptype('gsdgdf');

/* -------------------- SETS -------------------- */

SELECT SADD('set1', 'elem1');
SELECT SCARD('set1'); -- 1
SELECT SPTYPE('set1');
SELECT SADD('set1', 'elem2');
SELECT SADD('set1', 'elem2'); --duplicate
SELECT SADD('set1', 'elem3');
SELECT SCARD('set1'); -- 3

SELECT DEL('set1');
SELECT SCARD('set1');

SELECT SADD('set2', 'elem1');
SELECT SCARD('set2'); -- 1
SELECT SPTYPE('set2');
SELECT SADD('set2', 'elem2');
SELECT SADD('set2', 'elem2'); --duplicate
SELECT SADD('set2', 'elem3');
SELECT SCARD('set2'); -- 3
SELECT SPTYPE('set2');

SELECT SCARD('setxxxx');

SELECT SADD('set1', 'elem1'); -- remember we DEL set1 above
SELECT SISMEMBER('set1', 'elem1'); -- t
SELECT SISMEMBER('set1', 'elem4'); -- f
SELECT SISMEMBER('setxxxx', 'elem4'); -- f

-- SREM
SELECT SADD('remset1', 'elem1');
SELECT SCARD('remset1'); -- 1
SELECT SREM('remset1', 'nothere'); -- f
SELECT SISMEMBER('remset1', 'elem1'); -- t
SELECT SREM('remset1', 'elem1'); -- t
SELECT SISMEMBER('remset1', 'elem1'); -- f
SELECT SCARD('remset1'); -- 0

-- SPGET for set
SELECT SADD('set1', 'elem1');
SELECT SADD('set1', 'elem2');

SELECT SPGET('set1');

/* -------------------- LISTS -------------------- */

/* LPUSH -> LEN -> LPOP */
SELECT LPUSH('list1', 'elem1');
SELECT LLEN('list1'); -- 1
SELECT LPUSH('list1', 'elem2');
SELECT LLEN('list1'); -- 2

SELECT SPGET('list1');

SELECT LPOP('list1'); -- elem2
SELECT LLEN('list1'); -- 1

SELECT LPOP('list1'); -- elem1
SELECT LLEN('list1'); -- 0

SELECT LPOP('list1'); -- null
SELECT LLEN('list1'); -- 0

SELECT LPOP('list1'); -- null
SELECT LLEN('list1'); -- 0

SELECT SPGET('list1');

/* -------------------- HASHES -------------------- */

SELECT HSET('h1', 'f1', 'Hello');
SELECT HSET('h1', 'f2', 'World');

SELECT HGET('h1', 'f1'); -- Hello
SELECT HGET('h1', 'f2'); -- World
SELECT HGET('h1', 'f3'); -- NULL


SELECT HGET('h3', 'f1'); -- NULL

