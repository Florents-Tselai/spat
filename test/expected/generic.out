SET client_min_messages = ERROR;
SELECT spat_db_name(); -- spat-default
 spat_db_name 
--------------
 spat-default
(1 row)

-- switch to a new db
SET spat.db = 'test-spat';
SELECT spat_db_name(); -- test-spat
 spat_db_name 
--------------
 test-spat
(1 row)

SELECT spat_db_created_at() NOTNULL;
 ?column? 
----------
 t
(1 row)

-- switch back to the default to preserve my sanity when copy-pasting
SET spat.db = 'spat-default';
/* -------------------- STRINGS / GENERIC -------------------- */
-- SET - DELETE - SIZE
SELECT spset('tkey1', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT spset('tkeygdfg1', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT spset('gdf', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT spset('gfd', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT sp_db_nitems(); --4
 sp_db_nitems 
--------------
            4
(1 row)

-- DELETE and inspect size
SELECT del('tkey1'); --existing
 del 
-----
 t
(1 row)

SELECT del('aaaa'); --not existing
 del 
-----
 f
(1 row)

SELECT sp_db_nitems(); --3
 sp_db_nitems 
--------------
            3
(1 row)

SELECT del('tkeygdfg1');
 del 
-----
 t
(1 row)

SELECT del('gdf');
 del 
-----
 t
(1 row)

SELECT del('gfd');
 del 
-----
 t
(1 row)

SELECT sp_db_nitems(); --0
 sp_db_nitems 
--------------
            0
(1 row)

-- SET / GET / DEL text
SELECT spset('key1', 'value1');
 spset  
--------
 value1
(1 row)

SELECT spget('key1');
 spget  
--------
 value1
(1 row)

SELECT del('key1');
 del 
-----
 t
(1 row)

SELECT spget('key1');
 spget 
-------
 
(1 row)

-- TTL
SELECT SPSET('expkey1', 'expvalue1', ttl=> '1 second');
   spset   
-----------
 expvalue1
(1 row)

SELECT ttl('expkey1') < '1 second' ;
 ?column? 
----------
 t
(1 row)

-- DB SIZE BYTES
SELECT sp_db_size_bytes();
 sp_db_size_bytes 
------------------
          1048576
(1 row)

SELECT sp_db_size();
 sp_db_size 
------------
 1024 kB
(1 row)

-- sptype
SELECT spset('key1', 'value1');
 spset  
--------
 value1
(1 row)

SELECT sptype('key1');
 sptype 
--------
 string
(1 row)

SELECT sptype('gsdgdf');
 sptype 
--------
 null
(1 row)

/* -------------------- SETS -------------------- */
SELECT SADD('set1', 'elem1');
 sadd 
------
 
(1 row)

SELECT SCARD('set1'); -- 1
 scard 
-------
     1
(1 row)

SELECT SPTYPE('set1');
 sptype 
--------
 set
(1 row)

SELECT SADD('set1', 'elem2');
 sadd 
------
 
(1 row)

SELECT SADD('set1', 'elem2'); --duplicate
 sadd 
------
 
(1 row)

SELECT SADD('set1', 'elem3');
 sadd 
------
 
(1 row)

SELECT SCARD('set1'); -- 3
 scard 
-------
     3
(1 row)

SELECT DEL('set1');
 del 
-----
 t
(1 row)

SELECT SCARD('set1');
 scard 
-------
      
(1 row)

SELECT SADD('set2', 'elem1');
 sadd 
------
 
(1 row)

SELECT SCARD('set2'); -- 1
 scard 
-------
     1
(1 row)

SELECT SPTYPE('set2');
 sptype 
--------
 set
(1 row)

SELECT SADD('set2', 'elem2');
 sadd 
------
 
(1 row)

SELECT SADD('set2', 'elem2'); --duplicate
 sadd 
------
 
(1 row)

SELECT SADD('set2', 'elem3');
 sadd 
------
 
(1 row)

SELECT SCARD('set2'); -- 3
 scard 
-------
     3
(1 row)

SELECT SPTYPE('set2');
 sptype 
--------
 set
(1 row)

SELECT SCARD('setxxxx');
 scard 
-------
      
(1 row)

SELECT SADD('set1', 'elem1'); -- remember we DEL set1 above
 sadd 
------
 
(1 row)

SELECT SISMEMBER('set1', 'elem1'); -- t
 sismember 
-----------
 t
(1 row)

SELECT SISMEMBER('set1', 'elem4'); -- f
 sismember 
-----------
 f
(1 row)

SELECT SISMEMBER('setxxxx', 'elem4'); -- f
 sismember 
-----------
 f
(1 row)

-- SREM
SELECT SADD('remset1', 'elem1');
 sadd 
------
 
(1 row)

SELECT SCARD('remset1'); -- 1
 scard 
-------
     1
(1 row)

SELECT SREM('remset1', 'nothere'); -- f
 srem 
------
    0
(1 row)

SELECT SISMEMBER('remset1', 'elem1'); -- t
 sismember 
-----------
 t
(1 row)

SELECT SREM('remset1', 'elem1'); -- t
 srem 
------
    1
(1 row)

SELECT SISMEMBER('remset1', 'elem1'); -- f
 sismember 
-----------
 f
(1 row)

SELECT SCARD('remset1'); -- 0
 scard 
-------
     0
(1 row)

-- SPGET for set
SELECT SADD('set1', 'elem1');
 sadd 
------
 
(1 row)

SELECT SADD('set1', 'elem2');
 sadd 
------
 
(1 row)

SELECT SPGET('set1');
  spget  
---------
 set (2)
(1 row)

/* -------------------- LISTS -------------------- */
/* LPUSH -> LEN -> LPOP */
SELECT LPUSH('list1', 'elem1');
 lpush 
-------
 
(1 row)

SELECT LLEN('list1'); -- 1
 llen 
------
    1
(1 row)

SELECT LPUSH('list1', 'elem2');
 lpush 
-------
 
(1 row)

SELECT LLEN('list1'); -- 2
 llen 
------
    2
(1 row)

SELECT LPOP('list1'); -- elem2
 lpop  
-------
 elem2
(1 row)

SELECT LLEN('list1'); -- 1
 llen 
------
    1
(1 row)

SELECT LPOP('list1'); -- elem1
 lpop  
-------
 elem1
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

SELECT LPOP('list1'); -- null
 lpop 
------
 
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

SELECT LPOP('list1'); -- null
 lpop 
------
 
(1 row)

SELECT LLEN('list1'); -- 0
 llen 
------
    0
(1 row)

